### ISSUE TITLE
Strange "-f" issue

### ISSUE THREAD

**Issue by josephomorrow** [rg3/youtube-dl#7801]
When attempting to download certain videos, I get this weird error:

ERROR: The first format must contain the video, try using "-f 140+258"

(See attachment below):

[youtube-dl_20151208.txt](https://github.com/rg3/youtube-dl/files/55943/youtube-dl_20151208.txt)

**Comment by dstftw** [comment]
```
140          m4a        audio only DASH audio  127k , m4a_dash container, aac  @128k (44100Hz), 675.21KiB
258          mp4        audio only  386k , mp4a.40.2, 2.00MiB (best)
```
Both formats audio only. First must be video, second audio.

**Comment by josephomorrow** [comment]
Hello dstftw,

Apparently you didn't look at the command line I issued. I didn't enter "-f 140+258" or "-f 258+140"! What I entered was "-f mp4+m4a" (see script listing near top of my attachment).

Thanks for not being hasty,

Joseph

**Comment by josephomorrow** [comment]
Please see the new comment I just added to the "closed" issue. Apparently the parser sometimes reads my "mp4" to mean "audio only" instead of "video only". On 12/08/15, Sergey M.<notifications@github.com> wrote: Closed #7801.—Reply to this email directly or view it on GitHub.

**Comment by josephomorrow** [comment]
For video "https://www.youtube.com/watch?v=WiNuqUEAVDU"I get this using the "-F" option:[youtube] WiNuqUEAVDU: Downloading webpage[youtube] WiNuqUEAVDU: Downloading video info webpage[youtube] WiNuqUEAVDU: Extracting video information[youtube] WiNuqUEAVDU: Downloading DASH manifest[youtube] WiNuqUEAVDU: Downloading DASH manifest[info] Available formats for WiNuqUEAVDU:format code  extension  resolution note140          m4a        audio only DASH audio   96k , m4a_dash container, aac  @128k (44100Hz), 6.65MiB160          mp4        256x144    DASH video  112k , avc1.42c00c, 14fps, video only, 7.66MiB133          mp4        426x240    DASH video  251k , avc1.4d4015, 27fps, video only, 17.16MiB134          mp4        640x360    DASH video  603k , avc1.4d401e, 27fps, video only, 19.40MiB17           3gp        176x144    small ,  mp4a.40.2, mp4v.20.336           3gp        320x240    small ,  mp4a.40.2, mp4v.20.35            flv        400x240    small 43           webm       640x360    medium ,  vorbis, vp8.018           mp4        640x360    medium ,  mp4a.40.2, avc1.42001E (best)(Note that mp4 translates to "video"!)However: for video "https://www.youtube.com/watch?v=z5wX8jdEflY"I get this using the "-F" option:[youtube] z5wX8jdEflY: Downloading webpage[youtube] z5wX8jdEflY: Downloading video info webpage[youtube] z5wX8jdEflY: Extracting video information[youtube] z5wX8jdEflY: Downloading DASH manifest[youtube] z5wX8jdEflY: Downloading DASH manifest[info] Available formats for z5wX8jdEflY:format code  extension  resolution note249          webm       audio only DASH audio   50k , opus @ 50k, 238.80KiB250          webm       audio only DASH audio   72k , opus @ 70k, 319.45KiB171          webm       audio only DASH audio   96k , vorbis@128k (44100Hz), 445.46KiB140          m4a        audio only DASH audio  127k , m4a_dash container, aac  @128k (44100Hz), 675.21KiB251          webm       audio only DASH audio  137k , opus @160k, 576.77KiB278          webm       256x144    DASH video   94k , webm container, vp9, 15fps, video only, 388.53KiB242          webm       426x240    DASH video  110k , vp9, 30fps, video only, 398.02KiB160          mp4        256x144    DASH video  119k , avc1.42c00c, 15fps, video only, 585.79KiB243          webm       640x360    DASH video  206k , vp9, 30fps, video only, 777.95KiB133          mp4        426x240    DASH video  280k , avc1.4d4015, 30fps, video only, 1.29MiB134          mp4        640x360    DASH video  310k , avc1.4d401e, 30fps, video only, 1.38MiB244          webm       854x480    DASH video  326k , vp9, 30fps, video only, 1.25MiB135          mp4        854x480    DASH video  664k , avc1.4d401f, 30fps, video only, 2.85MiB247          webm       1280x720   DASH video  696k , vp9, 30fps, video only, 2.78MiB136          mp4        1280x720   DASH video 1341k , avc1.4d401f, 30fps, video only, 5.66MiB248          webm       1920x1080  DASH video 1605k , vp9, 30fps, video only, 6.33MiB137          mp4        1920x1080  DASH video 2502k , avc1.640028, 30fps, video only, 10.98MiB17           3gp        176x144    small ,  mp4a.40.2, mp4v.20.336           3gp        320x240    small ,  mp4a.40.2, mp4v.20.35            flv        400x240    small 43           webm       640x360    medium ,  vorbis, vp8.018           mp4        640x360    medium ,  mp4a.40.2, avc1.42001E22           mp4        1280x720   hd720 ,  mp4a.40.2, avc1.64001F256          mp4        audio only  194k , mp4a.40.5, 1.01MiB258          mp4        audio only  386k , mp4a.40.2, 2.00MiB (best)(Why does this choose an "audio only" as "best video"?) On 12/08/15, Sergey M.<notifications@github.com> wrote: 140          m4a        audio only DASH audio  127k , m4a_dash container, aac  @128k (44100Hz), 675.21KiB258          mp4        audio only  386k , mp4a.40.2, 2.00MiB (best)Both formats audio only. First must be video, second audio.—Reply to this email directly or view it on GitHub.
